src/hotspot/share/gc/g1/g1EvacFailureRegions.hpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2021, Huawei Technologies Co. Ltd. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_GC_G1_G1EVACFAILUREREGIONS_HPP
+#define SHARE_GC_G1_G1EVACFAILUREREGIONS_HPP
+
+#include "utilities/bitMap.inline.hpp"
```

Comment:
need "runtime/atomic.hpp" too (or so) for the `Atomic` operations uses.

Reply:
We do not allow includes of `.inline.hpp` files in `.hpp` files to avoid issues with circular dependencies. The `record` method needs to be put into an `.inline.hpp` file due to the use of `par_set_bit`.

Sorry for not noticing earlier.